UI/backend/app.py

```diff
@@ -11,106 +15,110 @@
-
+        return jsonify({'success': False, 'message': 'Instructor is required'}), requests.codes.bad_request
+    insert_func_args = [inst]
+    if course:
+        insert_func_args.append(course)
+        insert_course_args = [course]
+    if not course and time:
+        return jsonify({'success': False, 'message': 'If time is inputed course is mandatory'}), requests.codes.bad_request
+    if time:
+        ts_members = time.split()
+        if len(ts_members) != 3:
+            return jsonify({'success': False, 'message': 'Invalid time slot format'}), requests.codes.bad_request
+        day, start_time, end_time = ts_members
+        insert_func_args.extend([day, start_time, end_time])
+        insert_course_args.extend([day, start_time, end_time])
+        insert_time_slot(day, start_time, end_time)
     try:
-        if not course and not time:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst], capture_output=True, text=True)
-        elif not time and course:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst, course], capture_output=True, text=True)
-        elif time and course:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst, course, time], capture_output=True, text=True)
-        else:
-            return jsonify({'success': False, 'message': 'If time is inputed course is mandatory'}), 400
-        if result.returncode == 0:
-            return jsonify({'success': True, 'message': 'Instructor added successfully!'})
-        else:
-            return jsonify({'success': False, 'message': 'Failed to add instructor', 'details': result.stderr}), 500
-    except Exception as e:
-        print(f"Exception occurred: {str(e)}")
-        return jsonify({'success': False, 'message': str(e)}), 500
+        insert_instructor(*insert_func_args)
+        insert_course(*insert_course_args)
+        return jsonify({'success': True, 'message': 'Instructor added successfully!'})
+    except Exception as db_error:
+        return jsonify({'success': False, 'message': str(db_error)}), requests.codes.internal_server_error
 
 @app.route('/schedule', methods=['GET'])
 def schedule():
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
-
+    conn = connect_to_db()
+    cur = conn.cursor()
     try:
-        result = subprocess.run(['bash', cpp_exec_path, '--schedule'], capture_output=True, text=True)
+        cur.execute("SELECT json_build_object('courses', json_agg(row_to_json(t))) FROM (SELECT course_name, course_day, course_start_time, course_end_time FROM course) t")
+        courses_result = cur.fetchone()[0]
+        cur.execute("SELECT json_build_object('instructors', json_agg(row_to_json(t))) FROM (SELECT instructor_name, inst_course_name, inst_day, inst_start_time, inst_end_time FROM instructor) t")
```

Comment:
too long line. MAX LINE LENGTH should be 100

Reply:
change in other places if there are such

Reply:
done